test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlCustomQueryTests.cs

```diff
@@ -86,21 +95,19 @@ private async Task<bool> CheckIfSprocUsed(string hash)
                 _output.WriteLine("Checking database for sproc being run.");
                 conn.Open();
                 var cmd = conn.CreateCommand();
-                cmd.CommandText = "SELECT SCHEMA_NAME(sysobject.schema_id), OBJECT_NAME(stats.object_id),  stats.last_execution_time\r\n" +
-                    "FROM sys.dm_exec_procedure_stats stats\r\n" +
-                    "INNER JOIN sys.objects sysobject ON sysobject.object_id = stats.object_id\r\n" +
-                    "WHERE  sysobject.type = 'P'\r\n" +
-                    "and (sysobject.object_id = object_id('dbo.CustomQuery_" + hash + "') \r\n" +
-                    "OR sysobject.name = 'CustomQuery_" + hash + "')\r\n" +
-                    "ORDER BY stats.last_execution_time DESC  ";
+                cmd.CommandText = @"
+SELECT TOP 1 O.name
+  FROM sys.dm_exec_procedure_stats S
+       JOIN sys.objects O ON O.object_id = S.object_id
+  WHERE O.type = 'p' AND O.name = 'CustomQuery_'+@hash
+  ORDER BY
+       S.last_execution_time DESC";
+                cmd.Parameters.AddWithValue("@hash", hash);
+
                 var reader = cmd.ExecuteReader();
                 while (reader.Read())
```

Comment:
Does this still make sense to be in a while loop? This looks like just an if statement.

Reply:
It is a standard reader logic. I don't want to change it.